sync/src/block/downloader/body.rs

```diff
@@ -55,11 +55,15 @@ impl BodyDownloader {
     pub fn import_bodies(&mut self, hashes: Vec<BlockHash>, bodies: Vec<Vec<UnverifiedTransaction>>) {
         for (hash, body) in hashes.into_iter().zip(bodies) {
             if self.downloading.remove(&hash) {
+                let target = self.targets.iter().find(|t| t.hash == hash).expect("Downloading target must exist");
```

Comment:
I have a concern that this line of code will lookup n^2 of elements.
What do you think about creating `let target_map: HashMap<H256, Target>` before the for loop.

Reply:
Looks good. I'll fix it.

Reply:
Since it only checks whether the target is empty or not, I made a set of targets having an empty body.